src/networkParams.ts

```diff
@@ -63,31 +59,57 @@ export class TransactionOptions {
     }
 
     /**
-     * Creates a TransactionOptions object with the default options setting
+     * Creates a TransactionOptions object with the default options.
      */
-    static withDefaultOptions(): TransactionOptions {
+    static withDefaultOptions() {
         return new TransactionOptions(TRANSACTION_OPTIONS_DEFAULT);
     }
 
     /**
-     * Creates a TransactionsOptions object with the setting for hash signing
+     * Creates a TransactionOptions object from a set of options.
+     */
+    public static withOptions(options: {
+        withHashSign?: boolean,
+        withGuardian?: boolean
+    }): TransactionOptions {
+        let value = 0;
+
+        if (options.withHashSign) {
+            value |= TRANSACTION_OPTIONS_TX_HASH_SIGN;
+        }
+        if (options.withGuardian) {
+            value |= TRANSACTION_OPTIONS_TX_GUARDED;
+        }
+
+        return new TransactionOptions(value);
+    }
+
+    /**
+     * Returns true if the "hash sign" option is set.
+     */
+    isWithHashSign(): boolean {
+        return (this.value & TRANSACTION_OPTIONS_TX_HASH_SIGN) == TRANSACTION_OPTIONS_TX_HASH_SIGN;
+    }
+
+    /**
+     * Returns true if the "guarded transaction" option is set.
      */
-    static withTxHashSignOptions(): TransactionOptions {
```

Comment:
Or should we leave it, but as deprecated?

Reply:
I would remove it.

Reply:
Removed.